price-feeder/oracle/oracle.go

```diff
@@ -356,6 +370,54 @@ func SetProviderTickerPricesAndCandles(
 	return pricesOk || candlesOk
 }
 
+// Update update the values inside the OnChainData struct.
+func (onChainData *OnChainData) Update(currentBlockHeigh uint64, params oracletypes.Params) {
+	onChainData.lastUpdatedBlock = currentBlockHeigh
+	onChainData.params = &params
+}
+
+func (onChainData *OnChainData) IsParamsOutdated(currentBlockHeigh uint64) bool {
+	// doesn't have any data
+	if onChainData.params == nil {
+		return true
+	}
+
+	// state in which the blockchain is under cacheOnChainBlockQuantity
+	// amount of blocks, but already has value for params
+	if currentBlockHeigh < cacheOnChainBlockQuantity {
+		return false
+	}
+
+	// an edge case, should never happen
+	// which the current blockchain height is lower
+	// than the last updated block, to fix should
+	// just update the on chain params again
+	if currentBlockHeigh < onChainData.lastUpdatedBlock {
+		return true
+	}
+
+	return (currentBlockHeigh - onChainData.lastUpdatedBlock) > cacheOnChainBlockQuantity
+}
+
+// GetCachedParams returns the last updated parameters of the x/oracle module
+// if the current oracleOnChainParams is outdated we will query it again to
+// get the current oracle params.
+func (o *Oracle) GetCachedParams(currentBlockHeigh uint64) (oracletypes.Params, error) {
+	if !o.oracleOnChain.IsParamsOutdated(currentBlockHeigh) {
+		return *o.oracleOnChain.params, nil
```

Comment:
Why do we use a pointer here? Can we just remove all the reference & dereferencers?

Reply:
was to be able to check if `o.oracleOnChain.params == nil`

Reply:
![image](https://user-images.githubusercontent.com/17556614/175108049-64a5efe8-d69a-4fa8-a5ff-5b33676c3042.png)
Go didn't liked

Reply:
What do you think about using this way? 
![image](https://user-images.githubusercontent.com/17556614/175109150-1aaf2968-f663-4a1b-84d9-f52715335627.png)

Reply:
@adamewozniak I am going to ship as pointer right now